kola/tests/kubernetes/kubelet.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2018 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/platform/conf"
+	"github.com/coreos/mantle/util"
+)
+
+var (
+	staticPodConfig = conf.Ignition(`{
+    "ignition": {
+        "config": {},
+        "security": {
+            "tls": {}
+        },
+        "timeouts": {},
+        "version": "2.2.0"
+    },
+    "networkd": {},
+    "passwd": {},
+    "storage": {
+        "files": [
+            {
+                "contents": {
+                    "source": "data:,kind%3A%20KubeletConfiguration%0AapiVersion%3A%20kubelet.config.k8s.io%2Fv1beta1%0AclusterDomain%3A%20kube.local%0AclusterDNS%3A%20%5B10.254.0.10%5D%0AcgroupDriver%3A%20systemd%0AstaticPodPath%3A%20%2Fetc%2Fkubernetes%2Fmanifests%2F%0Aauthorization%3A%0A%20%20mode%3A%20AlwaysAllow%0Aauthentication%3A%0A%20%20anonymous%3A%0A%20%20%20%20enabled%3A%20true%0A%20%20webhook%3A%0A%20%20%20%20enabled%3A%20false%0A",
```

Comment:
Can you translate these into english for review at least?  @sjennings should review this, to make sure it looks like the bootstrap config we are going to use.

Reply:
```
kind: KubeletConfiguration
apiVersion: kubelet.config.k8s.io/v1beta1
clusterDomain: kube.local
clusterDNS: [10.254.0.10]
cgroupDriver: systemd
staticPodPath: /etc/kubernetes/manifests/
authorization:
  mode: AlwaysAllow
authentication:
  anonymous:
    enabled: true
  webhook:
    enabled: false
```

I noticed https://github.com/openshift/installer/blob/master/modules/ignition/resources/services/kubelet.service but kubelet in RHCOS is running 1.11 which uses a a kubeletconfiguration file for most of the options.  If there is already an existing kubeletconfiguration, let me know so I can use that instead of the one I created.

Reply:
```
kind: KubeletConfiguration
apiVersion: kubelet.config.k8s.io/v1beta1
clusterDomain: kube.local
clusterDNS: [10.254.0.10]
cgroupDriver: systemd
staticPodPath: /etc/kubernetes/manifests/
authorization:
  mode: AlwaysAllow
authentication:
  anonymous:
    enabled: true
  webhook:
    enabled: false
```
currently our manifest dir is `/etc/origin/node/pods`.  might change that to match.

Reply:
if 4.0 is switching to a kubelet, and the isntaller is going to use /etc/kubernetes/manifests like upstream, I think we should take this change to just abandon /etc/origin/node/pods....

Reply:
I'll keep it as is if that is the destination going forward.  Is there anything else that sticks out in terms of the kubeletconfiguration?  I created it based on the API with the least amount of options to get it working.